### PR TITLE
Add reference to scorecards

### LINKED ISSUE #14855: Add reference to OSSF Scorecards in the GitHub Actions hardening guide
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions

### What part(s) of the article would you like to see updated?

[OSSF Scorecards](https://github.com/ossf/scorecard) can find issues related to `token permissions` and `actions that are not pinned`. Adding a reference to Scorecards in the [token permissions](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#restricting-permissions-for-tokens) and (optionally) [pin actions](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions) sections of the hardening guide will help readers to take action to address these issues. 

https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by varunsh-coder**
### Why:

Closes #14855

[OSSF Scorecards](https://github.com/ossf/scorecard) can find issues related to `token permissions` and `actions that are not pinned`. This PR helps readers of the hardening guide to take action to address the issues. 
https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/

CC @josepalafox @laurentsimon

### What's being changed:

|**Source**|	**Staging**|	**Production**|	**What Changed** |
|-----------|-------------|-----------------|-----------------------|
|[content/actions/security-guides/security-hardening-for-github-actions.md](https://github.com/github/docs/blob/691b0d2d7846b469f5a43cf84e0f85a4b14383c0/content/actions/security-guides/security-hardening-for-github-actions.md)|	[Modified](https://docs-14856--patch-2.herokuapp.com/actions/security-guides/security-hardening-for-github-actions)|	[Original](https://docs.github.com/actions/security-guides/security-hardening-for-github-actions)|Added new section for OSSF Scorecards based on this [blog](https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/)

### Check off the following:

- [X] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@varunsh-coder 
Thanks so much for opening a PR! I'll get this triaged for review :zap: I edited your PR template slightly to properly link the issue and this PR 💖

**Comment by varunsh-coder** [comment]
I have added a new section for OpenSSF Scorecard

File: content/actions/security-guides/security-hardening-for-github-actions.md [hunk 1]
```diff
@@ -202,6 +202,10 @@ The same principles described above for using third-party actions also apply to
 {% data reusables.actions.outside-collaborators-internal-actions %} For more information, see "[Sharing actions and workflows with your enterprise](/actions/creating-actions/sharing-actions-and-workflows-with-your-enterprise)."
 {% endif %}
 
+## Using OpenSSF Scorecards to secure workflows
+
+Scorecards is an automated security tool that flags risky supply chain practices. You can use the Scorecards [GitHub Action](https://github.com/marketplace/actions/ossf-scorecard-action) and [starter workflow](https://github.com/actions/starter-workflows) to follow best security practices. Once configured, the Scorecards Action runs automatically on repository changes, and alerts developers about risky supply chain practices using the built-in code scanning experience. The Scorecards project makes a number of checks, including script injection attacks, token permissions, and pinned Actions.
```
Comment (ethomson) [inline content/actions/security-guides/security-hardening-for-github-actions.md:207]:
```suggestion
Scorecards is an automated security tool that flags risky supply chain practices. You can use the [Scorecards action](https://github.com/marketplace/actions/ossf-scorecard-action) and [starter workflow](https://github.com/actions/starter-workflows) to follow best security practices. Once configured, the Scorecards action runs automatically on repository changes, and alerts developers about risky supply chain practices using the built-in code scanning experience. The Scorecards project makes a number of checks, including script injection attacks, token permissions, and pinned actions.
```

We use little-a "actions" to refer to the composable units and big-A "Actions" to refer to the product.

**Review by ethomson — DISMISSED** [review]
Thanks for the PR, @varunsh-coder !

**Comment by varunsh-coder** [comment]
@ethomson Apologies for making more changes are PR approval. I got some feedback to make minor changes...

**Review by ethomson — APPROVED** [review]

**Review by ramyaparimi — APPROVED** [review]
@varunsh-coder Thanks so much for opening a PR with this fix! I'll get this merged once tests are passing 🍏

**Pull request merged**